app/routes/u.github/$slug.tsx

```diff
@@ -0,0 +1,121 @@
+import { json, LoaderFunction, redirect } from "@remix-run/node";
+import { Link, Outlet, useCatch, useLoaderData } from "@remix-run/react";
+import { FC } from "react";
+import RootLayout from "~/components/RootLayout";
+import { getUserProfileBySlug } from "~/database.server";
+import { UserProfile } from "~/types";
+
+export const loader: LoaderFunction = async ({ params }) => {
+  const slug = params.slug;
```

Comment:
Remix features nested routing, which is magical in the best way. This file is called `u.github/$slug.tsx`. Let's break it down:

- `u.github` creates 2 path segments: `/u/github`
- `$slug` is a dynamic path segment that accepts any string. Here, we expect a GitHub login

This means that if your GitHub login is "nova", your profile exists at `/u/github/nova`.

There's a 3rd thing to know about nested routing, but more on that later.